Realm/RLMRealmConfiguration+Sync.mm

```diff
@@ -51,6 +51,10 @@ - (void)setSyncConfiguration:(RLMSyncConfiguration *)syncConfiguration {
     self.config.in_memory = false;
     self.config.sync_config = std::make_shared<realm::SyncConfig>([syncConfiguration rawConfiguration]);
     self.config.schema_mode = realm::SchemaMode::Additive;
+    if (!self.config.encryption_key.empty()) {
+        self.config.sync_config.get()->realm_encryption_key = std::array<char, 64>();
```

Comment:
`.get()->` on a `unique_ptr` / `shared_ptr` is equivalent to just `->`.

I'd probably add `auto& sync_encryption_key = self.config.sync_config->realm_encryption_key;` here so that this line and the following can refer to it as `sync_encryption_key`.